File: backend/apps/academics/models/lesson.py
from django.db import models


class Lesson(models.Model):
    """Maps to `lesson` in db_current."""

    id = models.AutoField(primary_key=True)
    session_id = models.IntegerField(blank=False, null=False, db_index=True)
    subject_group_subject_id = models.IntegerField(
        blank=False, null=False, db_index=True
    )
    subject_group_class_sections_id = models.IntegerField(
        blank=False, null=False, db_index=True
    )
    name = models.CharField(max_length=255)
    created_at = models.DateTimeField()

    class Meta:
        managed = False
        db_table = "lesson"

    def __str__(self):
        return f"Lesson {self.pk}"

File: frontend/src/types/academics/lesson.ts
export interface Lesson {
  id: number;
  session_id: number;
  subject_group_subject_id: number;
  subject_group_class_sections_id: number;
  name: string;
  created_at: string;
  subject_name?: string | null;
  subject_group_name?: string | null;
  class_section_name?: string | null;
  subject_group_id?: number | null;
  subject_id?: number | null;
  class_section_id?: number | null;
}

export interface LessonCreatePayload {
  session_id: number;
  subject_group_id: number;
  subject_id: number;
  class_section_id: number;
  name: string;
}

export interface LessonUpdatePayload {
  session_id?: number;
  subject_group_id?: number;
  subject_id?: number;
  class_section_id?: number;
  name?: string;
}
